src/parser.rs

```diff
@@ -1010,9 +1010,23 @@ fn parse_variable_block(lexer: &mut ParseSession, linkage: LinkageType) -> Varia
 
     let access = parse_access_modifier(lexer);
 
-    let variables = parse_any_in_region(lexer, vec![KeywordEndVar], |lexer| {
+    let mut variables = parse_any_in_region(lexer, vec![KeywordEndVar], |lexer| {
         parse_variable_list(lexer)
     });
+
+    if constant {
```

Comment:
It feels to me that this fits better in the preprocessor, we have a mutable ast there we can edit, and we create new types or re-associate types there.

Reply:
we discussed in person following points:
- we agree that the preprocessor would be the right place
- the preprocessor is no full visitor, it mainly visits datatypes so I would need to build quite some visiting code to cover all the possible constants which adds the rist of me adding new bugs
- once we tackle a generic ast-visiting-pattern we're going to re-visit this, I created a tracking issue